File: src/FitnessTracker.Api/Models/BodyMeasurement.cs
namespace FitnessTracker.Api.Models;

public class BodyMeasurement
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid UserId { get; set; }

    public ApplicationUser User { get; set; } = null!;

    public DateTime RecordedAtUtc { get; set; } = DateTime.UtcNow;

    public double WeightKg { get; set; }

    public double? BodyFatPercentage { get; set; }

    public string? Notes { get; set; }
}

File: src/fitness-tracker-web/src/types/bodyMeasurement.ts
export interface BodyMeasurement {
    id: string;
    recordedAtUtc: string;
    weightKg: number;
    bodyFatPercentage: number | null;
    notes: string | null;
}

export interface CreateBodyMeasurementRequest {
    recordedAtUtc: string;
    weightKg: number;
    bodyFatPercentage: number | null;
    notes: string | null;
}

export interface UpdateBodyMeasurementRequest {
    recordedAtUtc: string;
    weightKg: number;
    bodyFatPercentage: number | null;
    notes: string | null;
}